ledger/src/blockstore_processor.rs

```diff
@@ -224,6 +225,8 @@ fn execute_batch(
         ..
     } = tx_results;
 
+    check_accounts_data_budget(&execution_results)?;
```

Comment:
What is the right place for this fn call? Should it be before `find_and_send_votes()` instead? Should it be below `send_transaction_status_batch()` instead?

Reply:
Below `find_and_send_votes` makes sense to me, since those we check those votes were executed correctly in that function `if tx.is_simple_vote_transaction() && result.was_executed_successfully() {`

It seems like it should be below `send_transaction_status_batch()` just to maintain parity with existing behavior. Right now we still send those transaction statuses even if there was a block error here: https://github.com/solana-labs/solana/blob/1e781127c38bca546b77d18571e10af61008c028/ledger/src/blockstore_processor.rs#L256-L257. Double checking with @CriesofCarrots, was this behavior (sending transaction statuses for erroring blocks) intentional?

Reply:
> It seems like it should be below `send_transaction_status_batch()` just to maintain parity with existing behavior. Right now we still send those transaction statuses even if there was a block error here:

Except if the block exceeds the max block cost limit, right?:
https://github.com/solana-labs/solana/blob/e87b941a5153d4ed5542e46826da352d2f7c065c/ledger/src/blockstore_processor.rs#L191-L212


---

I added the call to `check_accounts_data_budget()` above `send_transaction_status_batch()` just because that function _moves_ `execution_result`. So no reason for the current impl other than convenience.
https://github.com/solana-labs/solana/blob/e87b941a5153d4ed5542e46826da352d2f7c065c/ledger/src/blockstore_processor.rs#L238-L245